includes/classes/character.php

```diff
@@ -1,143 +1,125 @@
-
-                if(!isset($arr["type"])){
-                    if($arr["level"] > $group["level"]){
-                        $group["level"] = intval($arr["level"]);
-                    }
-                }
-
-                
-            }
-
-            if($group["level"] > 5){
-                $group["specialty"] = true;
-            }
-
-            array_push($aSkills, $group);
-
-            
-        }
-
-        return $aSkills;
-    }
-
-}
-?>
+<?php
+
+class character {
+
+	public function get_all() {
+		$stmt = database::$conn->prepare( "SELECT * FROM ecc_characters WHERE status NOT LIKE 'figurant%' AND sheet_status != 'deleted'" );
+		$res  = $stmt->execute();
+		$res  = $stmt->fetchAll( PDO::FETCH_ASSOC );
+		return $res;
+	}
+
+	public function get( $id, $needle ) {
+		if ( $needle == 'card_id' ) {
+			$stmt = database::$conn->prepare( "SELECT * FROM ecc_characters WHERE card_id = ? AND status NOT LIKE 'figurant%' AND sheet_status != 'deleted'" );
+			$res  = $stmt->execute( [ $id ] );
+			$res  = $stmt->fetch( PDO::FETCH_ASSOC );
+
+			if ( $res == null ) {
+				$s_hex = dechex( $id );
+				$a_dec = str_split( $s_hex, 2 );
+				if ( ! isset( $a_dec[1] ) ) {
+					return 'false';
+				}
+				$s_dec = '%' . $a_dec[3] . $a_dec[2] . $a_dec[1] . $a_dec[0] . '%';
+				if ( $s_dec == '%0%' ) {
+					return 'false';
+				}
+
+				$stmt = database::$conn->prepare( "SELECT * FROM ecc_characters WHERE card_id LIKE ? AND status NOT LIKE 'figurant%' AND sheet_status != 'deleted'" );
+				$res  = $stmt->execute( [ $s_dec ] );
+				$res  = $stmt->fetch( PDO::FETCH_ASSOC );
+			}
+		} elseif ( $needle == 'accountID' ) {
+			$stmt = database::$conn->prepare( "SELECT * FROM ecc_characters where $needle = ? AND status NOT LIKE 'figurant%' AND sheet_status != 'deleted'" );
+			$res  = $stmt->execute( [ $id ] );
+			$res  = $stmt->fetch( PDO::FETCH_ASSOC );
+		} else {
+			$stmt = database::$conn->prepare( "SELECT * FROM ecc_characters where $needle = ? AND status NOT LIKE 'figurant%' AND sheet_status != 'deleted'" );
+			$res  = $stmt->execute( [ $id ] );
+			$res  = $stmt->fetch( PDO::FETCH_ASSOC );
+		}
+
+		return $res;
+	}
+
+	function add_character( $account, $character ) {
+		// $check              = $this->check_card_id( $character['card_id'] );
+		$character_name     = $character['character_name'];
+		$card_id            = $character['card_id'];
+		$faction            = $character['faction'];
+		$rank               = $character['rank'];
+		$threat_assessment  = $character['threat_assessment'];
+		$douane_disposition = $character['douane_disposition'];
+		$douane_notes       = $character['douane_notes'];
+		$bastion_clearance  = $character['bastion_clearance'];
+		$icc_number         = $character['icc_number'];
+		$bloodtype          = $character['bloodtype'];
+		$ic_birthday        = $character['ic_birthday'];
+		$homeplanet         = $character['homeplanet'];
+		$status             = 'active';
+
+		// if ( ! $check ) {

```

Comment:
This commented line contains an open curcly bracket. Its close companion is still in here as well, but not commented out.